src/dialogs/generic/dialog-box.ts

```diff
@@ -26,7 +26,14 @@ class DialogBox extends LitElement {
 
   @query("ha-md-dialog") private _dialog?: HaMdDialog;
 
+  private _closePromise?: Promise<void>;
+
+  private _closeResolve?: () => void;
+
   public async showDialog(params: DialogBoxParams): Promise<void> {
+    if (this._closeState) {
```

Comment:
```suggestion
    if (this._closePromise) {
```
This is more clear imo